src/main/java/com/academy/task/repository/EmailRepository.java

```diff
@@ -0,0 +1,115 @@
+package com.academy.task.repository;
+
+import com.academy.task.model.Email;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Repository;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Repository
+public class EmailRepository {
```

Comment:
Create an interface for repository layer and rely on it in other places in the code. Make this class implement the interface. This will give you an opportunity to easily switch an implementation of the storage to database in future without touching other code